### ISSUE TITLE
Including jquery after $jaxon->getScript() breaks the jq function with no error output

### ISSUE THREAD

**Issue by walmartshopper** [jaxon-php/jaxon-core#146]
I have identified the issue I was having in jaxon 5 with some requests not working.  It appears if you include jquery in the page after calling $jaxon->getScript(), then calls to $response->jq() do not work.

For example if you have this in a page then calls to jq() in the response do not work:
```
<?php print $jaxon->getScript(true, true); ?>
<script src='https://code.jquery.com/jquery-3.7.1.min.js'></script>
```

However if you swap those two lines and include jquery first and then call $jaxon->getScript() second, then calls to jq() start working.

When they are in the wrong order, I did confirm using verbose debug that the jaxon request does return the correct commands.  However the calls to jq simply don't run, and there are no errors output in the browser's dev console.  It seems there might be some sort of conflict between chibi and jquery when chibi is included first, but since the dev console doesn't show any errors it's hard to know exactly what the problem is.

I can change the order of the includes and it fixes the problem, so I don't know if this is necessarily a bug.  However the lack of any errors made it difficult to troubleshoot, so it might be worth mentioning in the documentation that jquery should be included first before calling $jaxon->getScript()

Sample code to reproduce:
```
<?php
  require_once($_SERVER['DOCUMENT_ROOT']."/vendor/autoload.php");
  use Jaxon\Jaxon;
  use function Jaxon\jaxon;

  $jaxon = jaxon();
  $jaxon->setOption('js.app.minify', true);
  $jaxon->register(Jaxon::CALLABLE_FUNCTION, 'open_window');
  if($jaxon->canProcessRequest()) { $jaxon->processRequest(); }

  function open_window()
  {
    $response = jaxon()->newResponse();
    $response->jq('#dialog')->dialog('open');  // THIS REQUEST DOESNT WORK
    $response->assign("debug", "innerHTML", "test");  // THIS REQUEST WORKS
    return $response;
  }

?>
<html>
  <head>
    <link rel='stylesheet' href='https://code.jquery.com/ui/1.14.1/themes/base/jquery-ui.css'>
    <?php print $jaxon->getScript(true, true); ?>
    <script src='https://code.jquery.com/jquery-3.7.1.min.js'></script>
    <script src='https://code.jquery.com/ui/1.14.1/jquery-ui.min.js'></script>
  </head>
  <body>
    <script>
      $(function(){
        $("#dialog").dialog({
          title: "TEST DIALOG",
          modal: true,
          autoOpen: false
        });
      });
    </script>

    <button type="button" onclick="jaxon_open_window();">Open</button>
    <div id="dialog" style="display: none;">
      TEST DIALOG
    </div>
    <div id="debug">
    </div>
  </body>
</html>
```

**Comment by feuzeu** [comment]
Here's some tips about the v5 of the Jaxon library that I think will improve the example code.
You don't need to return the response object in the registered functions anymore, and you can take advantage of the [call factories](https://www.jaxon-php.org/docs/v5x/ui-features/call-factories.html) to set the button click handler.

```php
<?php
  require_once("../../vendor/autoload.php");
  use Jaxon\Jaxon;
  use function Jaxon\attr;
  use function Jaxon\jaxon;
  use function Jaxon\rq;

  $jaxon = jaxon();
  $jaxon->setOption('js.app.minify', true);
  $jaxon->register(Jaxon::CALLABLE_FUNCTION, 'open_window');
  if($jaxon->canProcessRequest()) { $jaxon->processRequest(); }

  function open_window()
  {
    $response = jaxon()->newResponse();
    $response->jq('#dialog')->dialog('open');  // THIS REQUEST DOESNT WORK
    $response->assign("debug", "innerHTML", "test");  // THIS REQUEST WORKS
  }

?>
<html>
  <head>
    <link rel='stylesheet' href='https://code.jquery.com/ui/1.14.1/themes/base/jquery-ui.css'>
    <?php print $jaxon->getScript(true, true); ?>
    <script src='https://code.jquery.com/jquery-3.7.1.min.js'></script>
    <script src='https://code.jquery.com/ui/1.14.1/jquery-ui.min.js'></script>
  </head>
  <body>
    <script>
      $(function(){
        $("#dialog").dialog({
          title: "TEST DIALOG",
          modal: true,
          autoOpen: false
        });
      });
    </script>

    <button type="button" <?= attr()->click(rq()->open_window()) ?>>Open</button>
    <div id="dialog" style="display: none;">
      TEST DIALOG
    </div>
    <div id="debug">
    </div>
  </body>
</html>
```

**Comment by walmartshopper** [comment]
Thanks, can confirm this is fixed.  I have a large codebase that was originally written for xajax and is now running on jaxon v4, but it's still using the older xajax style.  Migrating to jaxon v5 is proving to be difficult because I have hundreds of calls to $response->script() to deal with, so I might end up staying on v4 for a while.

**Comment by feuzeu** [comment]
Hi,
I've improved the solution to this issue a litte bit further.
The javascript library now will print a message in the console log if an error occurs while trying to execute a function from a server response.

**Comment by feuzeu** [comment]
> Migrating to jaxon v5 is proving to be difficult because I have hundreds of calls to $response->script() to deal with, so I might end up staying on v4 for a while.

Hi,

IMHO, having a large codebase is a good reason to migrate to Jaxon v5. Being able to attach UI components to your HTML nodes instead of assigning ids really makes a huge difference in that case. In addition to all the other improvements on request factories, commands and templating.

Of course some of the changes may require some amount of work, but I think it will be a big win on the long run for your app.

Also, I would like to let you know that I'm available to provide further support if you think it might help.
Just let me know so we can discuss about.

**Comment by walmartshopper** [comment]
Came across a couple more notes/questions while migrating to jaxon v5:

jaxon.ajax.callback.callback.onRequest = function() {} used to work for global callbacks but seems to be missing in v5.  Is there way to do global callbacks in v5 without migrating everything to the app structure?

Also I noticed that jaxon.getFormValues now submits values differently for multi-select inputs.  The values are now nested one level deeper in a sub-array as shown below.  Is this intentional?  If so, I have a lot of form processing code that will need to be updated to work with this structure.

v4:
```
(
  [input_name] => Array
  (
    [0] => value1
    [1] => value2
  )
)
```
v5:
```
(
  [input_name] => Array
  (
    [0] => Array
    (
      [0] => value1
      [1] => value2
    )
  )
)
```
